components/tikv_util/src/time.rs

```diff
@@ -405,8 +454,10 @@ impl SubAssign<Duration> for Instant {
 impl Sub<Instant> for Instant {
     type Output = Duration;
 
+    // TODO: For safety in production code, `sub` actually does saturating_sub.
+    // We should remove this operator from public scope.
```

Comment:
What places is using `sub`? Is it safe to do `saturating_duration_since` for all of them?

Reply:
I replaced them in the last commit, PTAL.

Reply:
Seems the trait implementation can be deleted.

Reply:
We are using it in "async_speed_limit::Limiter", I can make the op private though.